src/esphome/sensor/cse7766.cpp

```diff
@@ -39,34 +39,32 @@ void CSE7766Component::loop() {
   }
 }
 float CSE7766Component::get_setup_priority() const { return setup_priority::HARDWARE_LATE; }
+
 bool CSE7766Component::check_byte_() {
   uint8_t index = this->raw_data_index_;
   uint8_t byte = this->raw_data_[index];
   if (index == 0) {
-    // Header, usually 0x55, contains data about calibration etc.
-    // this is validated in parse_data_
-    return true;
+    if ((byte != 0x55) && ((byte & 0xF0) != 0xF0) && (byte != 0xAA)) {
+      return false;
+    }
   }
 
   if (index == 1) {
     if (byte != 0x5A) {
       ESP_LOGV(TAG, "Invalid Header 2 Start: 0x%02X!", byte);
       return false;
     }
-
-    return true;
```

Comment:
Re-add this line please. This function is intentionally laid out in a way that for each index the function will execute up to a certain point, not until the end of the function.